rero_ils/modules/holdings/api.py

```diff
@@ -310,16 +310,30 @@ def prediction_issues_preview(self, predictions=1):
         :param predictions: Number of the next issued to predict.
         :returns: An array of issues display text.
         """
-        if not self.patterns or not self.patterns.get('values'):
-            return []
-        patterns = deepcopy(self.patterns)
         text = []
-        for r in range(predictions):
-            text.append(self._get_next_issue_display_text(patterns))
-            patterns = self._increment_next_prediction(patterns)
+        if self.patterns and self.patterns.get('values'):
+            patterns = deepcopy(self.patterns)
+            for r in range(predictions):
+                text.append(self._get_next_issue_display_text(patterns))
+                patterns = self._increment_next_prediction(patterns)
         return text
 
 
+def prediction_issues_preview_for_pattern(predictions=1, patterns=None):
```

Comment:
can be a class method

Reply:
I would replace predictions by number_of_prediction or simply `n`.
I would also change the order the params as:
```
def prediction_issues_preview_for_pattern(patterns, predictions=1):
```
as patterns is required.